app/packages/dataset/src/Dataset.tsx

```diff
@@ -79,19 +97,22 @@ export const Dataset: React.FC<DatasetProps> = ({
   React.useLayoutEffect(() => {
     setCompactLayout(compactLayout);
   }, [compactLayout]);
-  React.useEffect(() => {
-    loadQuery({ name: dataset, savedViewSlug });
-  }, [dataset, savedViewSlug]);
-  React.useEffect(() => {
-    setCanChangeSavedViews(canEditSavedViews);
-  }, [canEditSavedViews]);
   React.useLayoutEffect(() => {
     setReadOnly(readOnly);
   }, [readOnly]);
   React.useLayoutEffect(() => {
     setTheme(theme);
   }, [theme]);
 
+  const context = useContext(fos.RouterContext);
+  const savedViewSlug = fos.getSavedViewName(context);
```

Comment:
Ideally there'd be a separate hook for it, but since this is something that runs in every render and seems to be sufficiently expensive, may I suggest you memoize it?

```
const savedViewSlug = React.useMemo(() => fos.getSavedViewName(context), [context]);
```

Reply:
updated